articles/rules-are-dangerous.md

```diff
@@ -0,0 +1,53 @@
+* The article by Martin Fowler about the “Page Object Pattern” once said to never put assertions into page objects. Someone read that article and wanted to convince the whole project team that it is very important to follow this rule very strictly. The arguments for the rule in the blog post were not very convincing to me and the benefit of it is at least very questionable for most applications of that pattern I have seen. But since Martin Fowler wrote this rule in his article, it counts as if it was written in the bible. This makes it difficult to argue and discuss constructively about it. No matter how many good examples you bring up in your context of application where the rule causes more troubles than bringing benefits, you will not be taken seriously. Such dogmatic discussions about such rules can cause you to lose valuable time and energy. A rule without good justification why it is so important and in what context, is not really a useful rule. No matter how famous and clever that person is who wrote it. Funny side note: in the meantime, the article by Fowler has been changed to mention both opinions on this topic and lets the option open to the reader to decide what fits best in his concrete context. Unfortunately, not all famous technical opinion makers are so wise as Mr. Fowler.
+
+Please, do not get me wrong: I like the Agile Manifesto, the SOLID Principles, Clean Code, Martin Fowler in general and even his article about the Page Object Pattern. Those are all practices I appreciate a lot and that influenced my way of thinking and working. But still all these things also come with the danger of undesired side-effects, if just following them stupidly or in a very dogmatic way.
+
+## The Problems with Rules
+
+The problems I often see in projects with such rules:
+* Applying rules in the wrong context might not have the desired effect at all
+* Using rules stricter than meant to be, causes adverse effects
+* Applying rules without really understanding them can result in more harm to the project than benefits
+* Some rules might be treated as god given, just because it has been defined in some famous article by some clever guy or in that holy Scrum bible or the process manual written by the higher management. This causes that people will not argue about these rules anymore about why, when and how some rules should be followed exactly. This can cause destructive discussion culture and lead to bad decisions without real arguments.
```

Comment:
```suggestion
* Treating rules as god given, causes people not to reason about these rules anymore. This can cause a destructive discussion culture and lead to bad decisions without real arguments.
```

Reply:
@abeggchr awesome - grossartig wie du das knackiger gemacht hast! Sehr cool getroffen! Danke!